### PR TITLE
Improve consensus in subround EndRound to avoid forks

### PR DESCRIPTION
**Pull request by SebastianMarian**
* Improve consensus in subround EndRound to avoid forks

**How to be tested**

**_CASE 1_**

Start a system test #allin with induced forks. After a while we should search and find this message in log files. 

_"doEndRoundJob: time is out -> cancel broadcasting final info and header"_

Also the hit rate should be much higher compared with other system tests ran with induced forks.

**_CASE 2_**

Start a system test #allin **WITHOUT** induced forks. After a while we should search in log files after this message: 

_"doEndRoundJob: time is out -> cancel broadcasting final info and header"_ . **THIS MESSAGE SHOULD NOT BE FOUND!**

### CODE REVIEW

**Review by raduchis — APPROVED** [review]

File: consensus/spos/bls/subroundEndRound.go [hunk 1]
```diff
@@ -204,6 +204,14 @@ func (sr *subroundEndRound) doEndRoundJobByLeader() bool {
 
 	// broadcast section
 
+	//TODO: Move this if before L205 before committing it, just to keep the induced hard forks system test functionality
+	if sr.Rounder().Index() > sr.RoundIndex {
+		log.Debug("doEndRoundJob: time is out -> cancel broadcasting final info and header",
```
Comment (iulianpascalau) [inline consensus/spos/bls/subroundEndRound.go:209]: I do not think this will work as expected because the rounder did not manage to increment its index.

Comment (SebastianMarian) [reply consensus/spos/bls/subroundEndRound.go:210]: Good catch, I will fix it

**Review by iulianpascalau — DISMISSED** [review]

**Review by sasurobert — APPROVED** [review]

File: consensus/spos/bls/subroundEndRound.go (hunk 1 above)
Comment (AdoAdoAdo) [reply consensus/spos/bls/subroundEndRound.go:210]: can we add also some tolerance?
we should test with some heavy load and make sure this does not stop the producing of blocks.

**Review by raduchis — APPROVED** [review]

**Review by sasurobert — APPROVED** [review]

**Review by AdoAdoAdo — APPROVED** [review]

**Review by gabi-vuls — APPROVED** [review]
System test passed.

**Pull request merged**